Assets/Plugins/Eskills/Android/EndgameActivity.kt

```diff
@@ -0,0 +1,102 @@
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultLauncher
+import android.app.Activity
+import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult
+import androidx.appcompat.app.AppCompatActivity
+import com.unity3d.player.UnityPlayer
+
+class EndgameActivity : AppCompatActivity() {
+    // activity result launcher
+    private lateinit var mLauncher: ActivityResultLauncher<Intent>
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val session = intent.getStringExtra(SESSION_KEY)
+        createLauncher()
+        launchEskillsEndgameFlow(session)
+    }
+
+    private fun createLauncher() {
+        mLauncher = registerForActivityResult(StartActivityForResult()) { result: ActivityResult ->
+            UnityPlayer.UnitySendMessage(
+                "Eskills",
+                "OnMatchCompleted",
+                result.resultCode.toString()
+            )
+            finish()
+        }
+    }
+
+    private fun launchEskillsEndgameFlow(session: String?) {
+        Log.d(TAG, "endgame session token : $session")
+        val url = ("https://apichain.catappult.io"
+            + "/transaction/eskills/endgame?"
+            + "session="
+            + session
+            + "&domain="
+            + applicationContext.packageName)
+        val intent = buildTargetIntent(url)
+        try {
+            mLauncher.launch(intent)
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+    }
+
+    /**
+    * This method generates the intent with the provided One Step URL to target the
+    * AppCoins Wallet.
+    *
+    * @param url The url that generated by following the One Step payment rules
+    *
+    * @return The intent used to call the wallet
+    */
+    private fun buildTargetIntent(url: String): Intent {
+        val intent = Intent(Intent.ACTION_VIEW)
+        intent.data = Uri.parse(url)
+        // Check if there is an application that can process the AppCoins Billing
+        // flow
+        val packageManager = applicationContext.packageManager
+        val appsList = packageManager.queryIntentActivities(intent, PackageManager.MATCH_DEFAULT_ONLY)
+        for (app in appsList) {
+            if (app.activityInfo.packageName == "cm.aptoide.pt") {
+                // If there's aptoide installed always choose Aptoide as default to open url
+                intent.setPackage(app.activityInfo.packageName)
+                break
+            } else if (app.activityInfo.packageName == "com.appcoins.wallet") {
+                // If Aptoide is not installed and wallet is installed then choose Wallet
+                // as default to open url
+                intent.setPackage(app.activityInfo.packageName)
+            }
+        }
+        return intent
+    }
```

Comment:
As this is repeated from PaymentActivity we should consider using hierarchy or at least add reusability